R/literal_coercion_linter.R

```diff
@@ -45,7 +53,7 @@ literal_coercion_linter <- function() {
       source_expression = source_expression,
       lint_message = paste(
         "Use literals directly where possible, instead of coercion.",
-        "c.f. 1L instead of as.integer(1), or NA_real_ instead of as.numeric(NA)."
+        "c.f. 1L instead of as.integer(1) or rlang::int(1), or NA_real_ instead of as.numeric(NA)."
```

Comment:
we should customize this to match what's actually linted. but it predates this PR, so we can put that as a follow-up issue

Reply:
Let's also try to fix this issue in the current PR.

How should we be customizing this? Any good example from existing linters for me draw inspiration?

Reply:
is it always possible to suggest an exact alternative? I think it should be. so

as.integer(1) -> suggest 1L
as.numeric(NA) -> suggest NA_real_

etc. looking again I think it's definitely a big enough undertaking to warrant its own PR

Reply:
Okay, sounds good. I can work on it in a different PR.

Reply:
#1441